xarray/backends/api.py

```diff
@@ -980,32 +1020,100 @@ def multi_file_closer():
     # read global attributes from the attrs_file or from the first dataset
     if attrs_file is not None:
         if isinstance(attrs_file, os.PathLike):
-            attrs_file = os.fspath(attrs_file)
+            attrs_file = cast(str, os.fspath(attrs_file))
         combined.attrs = datasets[paths.index(attrs_file)].attrs
 
     return combined
 
 
-WRITEABLE_STORES: Dict[str, Callable] = {
+WRITEABLE_STORES: dict[T_NETCDFENGINE, Callable] = {
     "netcdf4": backends.NetCDF4DataStore.open,
     "scipy": backends.ScipyDataStore,
     "h5netcdf": backends.H5NetCDFStore.open,
 }
 
 
+@overload
+def to_netcdf(
+    dataset: Dataset,
+    path_or_file: str | os.PathLike | None,
```

Comment:
Can this be `None` for this overload?

Reply:
I think so.
At least it is required to cover all possibilities for the overload and it seems to work (from my rough understanding of the code).